### PR TITLE
add baidu-qianfan llm

### PR DESCRIPTION
**Pull request by wencan**
# Description

This PR brings LLM support to Baidu Qianfan. Please allow me to delay adding examples.
https://qianfan.cloud.baidu.com/

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [x] Yes
- [ ] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [x] Yes
- [ ] No

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-integrations/llms/llama-index-llms-qianfan/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# LlamaIndex Llms Integration: Baidu Qianfan
+
+Baidu Intelligent Cloud's Qianfan LLM Platform Chat Client: Implementing the CustomLLM Abstraction Class.
```
Comment (logan-markewich) [inline llama-index-integrations/llms/llama-index-llms-qianfan/README.md:3]: we should add the install somewhere here: `pip install llama-index-llms-qianfan`

File: llama-index-integrations/llms/llama-index-llms-qianfan/llama_index/llms/qianfan/base.py [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
+        return gen()
+
+    @llm_chat_callback()
+    def complete(
+        self, prompt: str, formatted: bool = False, **kwargs: Any
+    ) -> CompletionResponse:
+        """
+        Not implemented.
+        """
+        raise NotImplementedError("complete is not supported for Qianfan-LLM")
```
Comment (logan-markewich) [inline llama-index-integrations/llms/llama-index-llms-qianfan/llama_index/llms/qianfan/base.py:367]: we actually have a wrapper to wrap a complete request into a chat request, you can see the OpenAI class for an example. Up to you though!

**Comment by wencan** [comment]
@logan-markewich 
The LLM developed by Baidu does not support the complete function.
I have added the ipynb and updated some comments.

**Comment by logan-markewich** [comment]
@wencan Yea Im saying `complete()` can call `chat()` -- this is exactly what openai does

**Comment by wencan** [comment]
@logan-markewich 
Ok, I have implemented four complete methods using the converter.

**Review by logan-markewich — APPROVED** [review]

**Comment by logan-markewich** [comment]
The publish might fail depending on the order -- will manually publish if so

**Pull request merged**